File: tools/bench/tasks/14_rect_lib.py
class Rect:
    def __init__(self, w, h):
        self.w = w
        self.h = h
def area(r):
    return r.w * r.h
def perim(r):
    return 2.0 * (r.w + r.h)
def scale(r, k):
    return Rect(r.w * k, r.h * k)
r = Rect(3.0, 4.0)
print(area(r))
print(perim(r))
print(area(scale(r, 2.0)))

File: tools/bench/answers/go_haiku/s1/14_rect_lib.go
package main

import "fmt"

type Rectangle struct {
	width  float64
	height float64
}

func area(r Rectangle) float64 {
	return r.width * r.height
}

func perimeter(r Rectangle) float64 {
	return 2 * (r.width + r.height)
}

func scale(r Rectangle, factor float64) Rectangle {
	return Rectangle{r.width * factor, r.height * factor}
}

func main() {
	r := Rectangle{3.0, 4.0}
	fmt.Println(area(r))
	fmt.Println(perimeter(r))
	r2 := scale(r, 2.0)
	fmt.Println(area(r2))
}
